src/gtc/passes/oir_pipeline.py

```diff
@@ -78,28 +93,21 @@ def steps(self) -> Sequence[PASS_T]:
             FillFlushToLocalKCaches,
         ]
 
-    def apply(self, steps: Sequence[PASS_T]) -> oir.Stencil:
-        result = self.oir
-        for step in steps:
-            if isinstance(step, type) and issubclass(step, NodeVisitor):
-                result = step().visit(result)
-            else:
-                result = step(result)
-        return result
-
-    def _get_cached(self, steps: Sequence[PASS_T]) -> Optional[oir.Stencil]:
-        return self._cache.get(tuple(hash_step(step) for step in steps))
+    @property
+    def steps(self) -> Sequence[PASS_T]:
+        hash_skips = {hash_step(step) for step in self.skip}
```

Comment:
I would remove the indirection via `hash_step`. We just require that steps need to be hashable. Or maybe I miss way hash_step exists.

Reply:
👍   I was also confused why it was there, but I kept it at first. I just deleted it.